producer/producer.go

```diff
@@ -172,3 +187,59 @@ func envelopeMessage(key string, message string) (string, error) {
 
 	return string(jsonRecords), err
 }
+
+// ConnectivityCheck verifies if the kakfa proxy is availabe
+func (p *DefaultMessageProducer) ConnectivityCheck() (string, error) {
+	err := p.checkMessageQueueProxyReachable()
+	if err == nil {
+		return "Connectivity to producer proxy is OK.", nil
+	}
+	log.Printf("ERROR - Producer Connectivity Check - %s", err)
+	return "Error connecting to producer proxy", err
+}
+
+func (p *DefaultMessageProducer) checkMessageQueueProxyReachable() error {
```

Comment:
Same as for the gonsumer, might be worth renaming this, or not treating the absence of the configured topic as an error, but just log it as WARN.